exercises/matrix/src/test/java/MatrixTest.java

```diff
@@ -0,0 +1,184 @@
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Enclosed.class)
+public class MatrixTest {
+
+    @RunWith(Parameterized.class)
+    public static class GetFirstRowTest {
+        private Matrix matrix;
+        private int[] firstRow;
+
+        @Parameterized.Parameters(name = "{index}: checking first row of matrix constructed with string \"{0}\".")
+        public static Collection<Object[]> data() {
+            return Arrays.asList(new Object[][]{
+                    {"0", new int[] {0}},
+                    {"0 1", new int[] {0, 1}},
+                    {"0\\n1", new int[] {0}},
```

Comment:
Ohh... is this for formatting the parametrized test output? If so, that's a tricky balance; I fear adding then adjusting the newlines in this way might cause implementors some confusion.